docs/en/concepts-and-designs/lal.md

```diff
@@ -0,0 +1,216 @@
+filter {
+    text {
+        regexp "(?<timestamp>\\d{8}) (?<thread>\\w+) (?<level>\\w+) (?<traceId>\\w+) (?<msg>.+)"
+        // this is just a demo pattern
+    }
+    extractor {
+        tag key: "level", val: parsed["level"]
+        // we add a tag called `level` and its value is parsed["level"], captured from the regexp above
+        traceId parsed["traceId"]
+        // we also extract the trace id from the parsed result, which will be used to associate the log with the trace
+    }
+    // ...
+}
+```
+
+- `grok`
+
+<!-- TODO: grok Java library has poor performance, need to benchmark it, the idea is basically the same with `regexp` above -->
+
+### Extractor
+
+Extractors aim to extract metadata from the logs. The metadata can be a service name, a service instance name, an
+endpoint name, or even a trace ID, all of which can be associated with the existing traces and metrics.
+
+- `service`
+
+`service` extracts the service name from the `parsed` result, and set it into the `LogData`, which will be persisted (if
+not dropped) and is used to associate with traces / metrics.
+
+- `instance`
+
+`instance` extracts the service instance name from the `parsed` result, and set it into the `LogData`, which will be
+persisted (if not dropped) and is used to associate with traces / metrics.
+
+- `endpoint`
+
+`endpoint` extracts the service instance name from the `parsed` result, and set it into the `LogData`, which will be
+persisted (if not dropped) and is used to associate with traces / metrics.
+
+- `traceId`
+
+`traceId` extracts the trace ID from the `parsed` result, and set it into the `LogData`, which will be
+persisted (if not dropped) and is used to associate with traces / metrics.
+
+- `segmentId`
+
+`segmentId` extracts the segment ID from the `parsed` result, and set it into the `LogData`, which will be
+persisted (if not dropped) and is used to associate with traces / metrics.
+
+- `spanId`
+
+`spanId` extracts the span ID from the `parsed` result, and set it into the `LogData`, which will be
+persisted (if not dropped) and is used to associate with traces / metrics.
+
+- `metrics`
+
+`metrics` extracts / generates metrics from the logs. The supported metrics are `CounterMetrics`, `GaugeMetrics`,
+and `HistogramMetrics`, examples are as follows:
+
+```groovy
+filter {
+    // ...
+    extractor {
+        service parsed["serviceName"]
+        metrics {
+            counter { // whenever the filter is fed with a piece of log, the counter will be increased by one
+                name: "logsCount"
+                tips: "The total count of received logs"
+                tags: ["key1": "value1", "key2": "value2"]
+            }
+            if (parsed["level"] == "ERROR") {
+                counter {
+                    // whenever the filter is fed with a piece of log whose level is ERROR, the counter will be increased by one
+                    name: "errorLogsCount"
+                    tips: "The total count of error logs"
```

Comment:
I think we don't have `tips` in the MAL?

Reply:
> I think we don't have `tips` in the MAL?

I plan to create the metrics in LAL via `org.apache.skywalking.oap.server.telemetry.api.MetricsCreator`, which have `tips` as a parameter

Reply:
That is a wrong direction. MetricsCreator you pointed, is using the OAP to measure metrics. If you take a closer look, you will see, we need OTEL or Prometheus fetcher to read the data back to MAL.
Instead, you should take a look how OC-receiver and Zabbix-receiver works, transfer the data into `SampleFamily`.

Reply:
> That is a wrong direction. MetricsCreator you pointed, is using the OAP to measure metrics. If you take a closer look, you will see, we need OTEL or Prometheus fetcher to read the data back to MAL.
> Instead, you should take a look how OC-receiver and Zabbix-receiver works, transfer the data into `SampleFamily`.

Make sense to me